server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMessagesUpdateProcessor.java

```diff
@@ -110,8 +127,23 @@ private void update(MessageId messageId, UpdateMessagePatch updateMessagePatch,
                 } else {
                     builder.updated(ImmutableList.of(messageId));
                 }
+                Stream<MailboxId> targetMailboxIds = listTargetMailboxIds(updateMessagePatch);
+                if (isTargetingOutbox(mailboxSession, targetMailboxIds)) {
+                    Optional<MessageResult> messagesToSend =
+                        messageIdManager.getMessages(
+                            ImmutableList.of(messageId), FetchGroupImpl.FULL_CONTENT, mailboxSession)
+                        .stream()
+                        .findFirst();
+                    if (messagesToSend.isPresent()) {
+                        MailImpl mail = MailImpl.fromMimeMessage(messageId.serialize(),
+                            new MimeMessage(Session.getDefaultInstance(new Properties()), messagesToSend.get().getFullContent().getInputStream()));
```

Comment:
That's uggly and incredibly low level!

I'd prefer having a mailImpl method taking an InputStream instead of "this".

Reply:
I partially agree but ... passing an InputStream is even more low level ... we have to choose between to evil

Reply:
Well, at least do that in a helper class ;-)